source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -0,0 +1,26 @@
+using F0.Minesweeper.Logic.Abstractions;
+
+namespace F0.Minesweeper.Logic
+{
+	public class Minefield : IMinefield
+	{
+		private readonly uint width, height, mineCount;
+
+		internal Minefield(uint width, uint height, uint mineCount)
+		{
+			this.width = width;
+			this.height = height;
+			this.mineCount = mineCount;
+		}
+
+		public IGameUpdateReport Uncover(uint x, uint y) => Uncover(new Location(x, y));
+		public IGameUpdateReport Uncover(ILocation location)
```

Comment:
I feel like there's a test missing for this. Yes, is bomb and number of adjacent bombs are completely mocked but at least we could test that uncover returns minimum the clicked location within the uncoveredcell array

Reply:
Yep, thiss is a really good idea, thanks for this suggestion 😄